open-hackathon-server/src/hackathon/template/template_unit.py

```diff
@@ -36,3 +36,12 @@ def __init__(self, provider):
         provider should be value of enum VE_PROVIDER.
         """
         self.provider = provider
+
+    def get_name(self):
+        raise NotImplemented
```

Comment:
see [`get_user_templates`](https://github.com/msopentechcn/open-hackathon/blob/master/open-hackathon-server/src/hackathon/hack/hackathon_template_manager.py#L89-L105)

this method require every template_units have a `get_name`, `get_type`, `get_description`, so I add the three methods as a abstraction.

but i cannot figure out the `name`,`type` and `description` for a `azure_template_unit`, because the old azure template didn't configure this